### PR TITLE
Upgrade to XCode 15

### PR DESCRIPTION
**Pull request by klochowicz**
- catch firebase exception if cannot fetch fcm_token
- fix build errors in XCode 15

### CODE REVIEW

File: mobile/lib/main.dart
Comment (bonomat) [inline mobile/lib/main.dart:557]: Given we send this token to the coordinator to register, should we set it here explicitly to something so that we know the user had an error and we could not get a token for them ?

**Comment by holzeis** [comment]
🚨 The changes introduced to support a xcode-15 build (6297a1e7f35be754a43d29e1f60588b690a4a83b) breaks the build for xcode-14 (Version 14.3.1)

In order for this commit to work you will have to upgrade your mac os to Ventura Version 13.6 and upgrade xcode to Version 15.0.

Note, I don't think we do have to upgrade to xcode-15. @klochowicz you could have unblocked yourself by running a separate xcode-14 installation.

```
cp target/universal/release/libnative.a mobile/ios/Runner
Launching lib/main.dart on iPhone 14 Pro in debug mode...
Running pod install...                                             490ms
CocoaPods' output:
↳
      Preparing

    Analyzing dependencies

    Inspecting targets to integrate
      Using `ARCHS` setting to build architectures of target `Pods-Runner`: (``)

    Finding Podfile changes
      - Flutter
      - firebase_core
      - firebase_messaging
      - flutter_local_notifications
      - flutter_native_splash
      - package_info_plus
      - path_provider_foundation
      - share_plus
      - shared_preferences_foundation
      - social_share
      - url_launcher_ios

    Fetching external sources
    -> Fetching podspec for `Flutter` from `Flutter`
    -> Fetching podspec for `firebase_core` from `.symlinks/plugins/firebase_core/ios`
    firebase_core: Using Firebase SDK version '10.12.0' defined in 'firebase_core'
    -> Fetching podspec for `firebase_messaging` from `.symlinks/plugins/firebase_messaging/ios`
    firebase_messaging: Using Firebase SDK version '10.12.0' defined in 'firebase_core'
    -> Fetching podspec for `flutter_local_notifications` from `.symlinks/plugins/flutter_local_notifications/ios`
    -> Fetching podspec for `flutter_native_splash` from `.symlinks/plugins/flutter_native_splash/ios`
    -> Fetching podspec for `package_info_plus` from `.symlinks/plugins/package_info_plus/ios`
    -> Fetching podspec for `path_provider_foundation` from `.symlinks/plugins/path_provider_foundation/darwin`
    -> Fetching podspec for `share_plus` from `.symlinks/plugins/share_plus/ios`
    -> Fetching podspec for `shared_preferences_foundation` from `.symlinks/plugins/shared_preferences_foundation/darwin`
    -> Fetching podspec for `social_share` from `.symlinks/plugins/social_share/ios`
    -> Fetching podspec for `url_launcher_ios` from `.symlinks/plugins/url_launcher_ios/ios`

    Resolving dependencies of `Podfile`
      CDN: trunk Relative path: CocoaPods-version.yml exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: all_pods_versions_0_3_5.txt exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: Specs/0/3/5/Firebase/10.13.0/Firebase.podspec.json exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: Specs/0/3/5/Firebase/10.12.0/Firebase.podspec.json exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: Specs/0/3/5/Firebase/10.12.0/Firebase.podspec.json exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: all_pods_versions_4_2_c.txt exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: Specs/4/2/c/FlutterMacOS/3.0.0/FlutterMacOS.podspec.json exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: all_pods_versions_8_b_d.txt exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: Specs/8/b/d/FirebaseCore/10.13.0/FirebaseCore.podspec.json exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: all_pods_versions_2_d_6.txt exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: Specs/2/d/6/FirebaseMessaging/10.13.0/FirebaseMessaging.podspec.json exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: Specs/8/b/d/FirebaseCore/10.12.0/FirebaseCore.podspec.json exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: all_pods_versions_0_8_4.txt exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: Specs/0/8/4/GoogleUtilities/7.11.5/GoogleUtilities.podspec.json exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: Specs/0/8/4/GoogleUtilities/7.11.5/GoogleUtilities.podspec.json exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: Specs/0/8/4/GoogleUtilities/7.11.5/GoogleUtilities.podspec.json exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: all_pods_versions_7_9_1.txt exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: Specs/7/9/1/FirebaseCoreInternal/10.13.0/FirebaseCoreInternal.podspec.json exists! Returning local because checking is only performed in repo update
    [!] CocoaPods could not find compatible versions for pod "FirebaseCoreInternal":
      In snapshot (Podfile.lock):
        FirebaseCoreInternal (= 10.15.0, ~> 10.0)

      In Podfile:
        firebase_core (from `.symlinks/plugins/firebase_core/ios`) was resolved to 2.15.0, which depends on
          Firebase/CoreOnly (= 10.12.0) was resolved to 10.12.0, which depends on
            FirebaseCore (= 10.12.0) was resolved to 10.12.0, which depends on
              FirebaseCoreInternal (~> 10.0)


    You have either:
     * out-of-date source repos which you can update with `pod repo update` or with `pod install --repo-update`.
     * changed the constraints of dependency `FirebaseCoreInternal` inside your development pod `firebase_core`.
       You should run `pod update FirebaseCoreInternal` to apply changes you've made.

    /opt/homebrew/Cellar/cocoapods/1.12.1/libexec/gems/molinillo-0.8.0/lib/molinillo/resolution.rb:317:in `raise_error_unless_state'
    /opt/homebrew/Cellar/cocoapods/1.12.1/libexec/gems/molinillo-0.8.0/lib/molinillo/resolution.rb:299:in `block in unwind_for_conflict'
    <internal:kernel>:90:in `tap'
    /opt/homebrew/Cellar/cocoapods/1.12.1/libexec/gems/molinillo-0.8.0/lib/molinillo/resolution.rb:297:in `unwind_for_conflict'
    /opt/homebrew/Cellar/cocoapods/1.12.1/libexec/gems/molinillo-0.8.0/lib/molinillo/resolution.rb:257:in `process_topmost_state'
    /opt/homebrew/Cellar/cocoapods/1.12.1/libexec/gems/molinillo-0.8.0/lib/molinillo/resolution.rb:182:in `resolve'
    /opt/homebrew/Cellar/cocoapods/1.12.1/libexec/gems/molinillo-0.8.0/lib/molinillo/resolver.rb:43:in `resolve'
    /opt/homebrew/Cellar/cocoapods/1.12.1/libexec/gems/cocoapods-1.12.1/lib/cocoapods/resolver.rb:94:in `resolve'
    /opt/homebrew/Cellar/cocoapods/1.12.1/libexec/gems/cocoapods-1.12.1/lib/cocoapods/installer/analyzer.rb:1082:in `block in resolve_dependencies'
    /opt/homebrew/Cellar/cocoapods/1.12.1/libexec/gems/cocoapods-1.12.1/lib/cocoapods/user_interface.rb:64:in `section'
    /opt/homebrew/Cellar/cocoapods/1.12.1/libexec/gems/cocoapods-1.12.1/lib/cocoapods/installer/analyzer.rb:1080:in `resolve_dependencies'
    /opt/homebrew/Cellar/cocoapods/1.12.1/libexec/gems/cocoapods-1.12.1/lib/cocoapods/installer/analyzer.rb:125:in `analyze'
    /opt/homebrew/Cellar/cocoapods/1.12.1/libexec/gems/cocoapods-1.12.1/lib/cocoapods/installer.rb:422:in `analyze'
    /opt/homebrew/Cellar/cocoapods/1.12.1/libexec/gems/cocoapods-1.12.1/lib/cocoapods/installer.rb:244:in `block in resolve_dependencies'
    /opt/homebrew/Cellar/cocoapods/1.12.1/libexec/gems/cocoapods-1.12.1/lib/cocoapods/user_interface.rb:64:in `section'
    /opt/homebrew/Cellar/cocoapods/1.12.1/libexec/gems/cocoapods-1.12.1/lib/cocoapods/installer.rb:243:in `resolve_dependencies'
    /opt/homebrew/Cellar/cocoapods/1.12.1/libexec/gems/cocoapods-1.12.1/lib/cocoapods/installer.rb:162:in `install!'
    /opt/homebrew/Cellar/cocoapods/1.12.1/libexec/gems/cocoapods-1.12.1/lib/cocoapods/command/install.rb:52:in `run'
    /opt/homebrew/Cellar/cocoapods/1.12.1/libexec/gems/claide-1.1.0/lib/claide/command.rb:334:in `run'
    /opt/homebrew/Cellar/cocoapods/1.12.1/libexec/gems/cocoapods-1.12.1/lib/cocoapods/command.rb:52:in `run'
    /opt/homebrew/Cellar/cocoapods/1.12.1/libexec/gems/cocoapods-1.12.1/bin/pod:55:in `<top (required)>'
    /opt/homebrew/Cellar/cocoapods/1.12.1/libexec/bin/pod:25:in `load'
    /opt/homebrew/Cellar/cocoapods/1.12.1/libexec/bin/pod:25:in `<main>'

Error output from CocoaPods:
↳
    [!] The version of CocoaPods used to generate the lockfile (1.13.0) is higher than the version of the current executable (1.12.1). Incompatibility issues may arise.

Error: CocoaPods's specs repository is too out-of-date to satisfy dependencies.
To update the CocoaPods specs, run:
  pod repo update



Error running pod install
Error launching application on iPhone 14 Pro.
error: Recipe `run` failed with exit code 1
```

**Review by holzeis — APPROVED** [review]
Thanks for the fix! It would have been great if you could have separated the commits into separate PRs.

Note, I just upgraded my local environment to Ventura Version 13.6, xcode to Version 15.0 and cocoapods to 1.13.0 and managed to compile that commit (https://github.com/get10101/10101/commit/6297a1e7f35be754a43d29e1f60588b690a4a83b).

Additionally I have an additional xcode version 14.2 installed allowing me work on branches which are not upgraded to xcode-15 yet. I propose to follow that practice of defining a fixed xcode version for the project non-the-less so that we do not run into this issue the next time.

**Comment by bonomat** [comment]
The firebase fix was cherry-picked into: https://github.com/get10101/10101/pull/1377#pullrequestreview-1644218188

**Comment by klochowicz** [comment]
> The firebase fix was cherry-picked into: #1377

awesome, thanks! I've only coupled them together as I could not have tested them separately :)

**Comment by klochowicz** [comment]
> 🚨 The changes introduced to support a xcode-15 build ([6297a1e](https://github.com/get10101/10101/commit/6297a1e7f35be754a43d29e1f60588b690a4a83b)) break the build for xcode-14 (Version 14.3.1)
> 
> In order for this commit to work you will have to upgrade your mac os to Ventura Version 13.6, upgrade xcode to Version 15.0 and cocoapods 1.13.0

The first one is not true, I'm running Ventura 13.5.2. 
I wonder whether XCode 15.0 would be needed too 🤔 certainly newer cocoapods, that's for sure. To be cautious, I'd assume both. 

> 
> // edit: You will also have to update cocoapods `brew upgrade cocoapods` and run `pod repo update` afterwards
> 


> Note, I don't think we do have to upgrade to xcode-15. @klochowicz you could have unblocked yourself by running a separate xcode-14 installation.

Tried and didn't succeed. IMHO we should keep compatibility with the latest release of Apple's build system, and I did not anticipate pushback received. This work has to be done sooner or later.

**Comment by klochowicz** [comment]
I've re-requested review from everyone , to be sure we're aligned when we decide to finally bump XCode & CocoaPods.

If possible, check whether this would work with *just* cocoapods change, because this would unblock me (otherwise, I'm stuck with locally applying the first commit to all my work and dropping it before pushing to branches; that's workable, but not optimal).

**Review by holzeis — APPROVED** [review]
> The first one is not true, I'm running Ventura 13.5.2.

It's true that you have to upgrade to Ventura as you otherwise can't update xcode to version 15. However the minor version can obviously deviate. Interestingly Sonoma 14.0 has just been release 😅 

> I've spent a considerable amount of time trying to make this work - I've put a XCode 14.2 into Applications folder too, and flutter doctor was adamant to keep choosing Xcode 15 😢

That's odd, but I remember also investing quite some time when I accidentally upgraded my xcode and failed to compile the project anymore. My solution was to install a second Xcode which is working for me ever since, but maybe I did some additional change that I can' t remember anymore.

> IMHO we should keep compatibility with the latest release of Apple's build system, and I did not anticipate pushback received. This work has to be done sooner or later.

No pushback from my side 😉. You already got my approval for your initial request. 👍 It's just that everybody developing on mac would have to do all of these upgrades mentioned above to work with xcode-15, which took me at least 2 hours. That's the only reason I raised a concern, but if everybody has upgraded their environments we are good to go 👍

**Comment by klochowicz** [comment]
> > The first one is not true, I'm running Ventura 13.5.2.
> 
> It's true that you have to upgrade to Ventura as you otherwise can't update xcode to version 15. However the minor version can obviously deviate. Interestingly Sonoma 14.0 has just been release 😅

I can confirm that Sonoma 14.0 works fine with 10101 :-) at least with XCode 15 :P 

> 
> > I've spent a considerable amount of time trying to make this work - I've put a XCode 14.2 into Applications folder too, and flutter doctor was adamant to keep choosing Xcode 15 😢
> 
> That's odd, but I remember also investing quite some time when I accidentally upgraded my xcode and failed to compile the project anymore. My solution was to install a second Xcode which is working for me ever since, but maybe I did some additional change that I can' t remember anymore.
> 
> > IMHO we should keep compatibility with the latest release of Apple's build system, and I did not anticipate pushback received. This work has to be done sooner or later.
> 
> No pushback from my side 😉. You already got my approval for your initial request. 👍 It's just that everybody developing on mac would have to do all of these upgrades mentioned above to work with xcode-15, which took me at least 2 hours. That's the only reason I raised a concern, but if everybody has upgraded their environments we are good to go 👍

fair enough! it takes a bit, but it's an upgrade that comes to us sooner or later. 

I'll wait for green light from @bonomat before merging though, just to be sure all mac people are covered! (perhaps a good time to upgrade at least XCode & cocoapods on a Friday arvo 😉)

**Comment by bonomat** [comment]
mhmm... I've upgraded and now I get an error (also on this branch)

```
Error (Xcode): DT_TOOLCHAIN_DIR cannot be used to evaluate LIBRARY_SEARCH_PATHS, use TOOLCHAIN_DIR instead
```

```
pod --version
1.13.0
```

image

Any thoughts?

**Review by bonomat — APPROVED** [review]
Let's just hope it also works on CI

**Comment by klochowicz** [comment]
I'll merge it, and check whether CI has worked with publishing the latest version to iOS. If it doesn't, I'll revert this PR.

**Comment by klochowicz** [comment]
> Let's just hope it also works on CI

Found an issue in fastlane - it's unfortunate that `riggrep` by default excludes stuff from hidden directories and that's where the CI actions are :( should be fixed with #1414

**Pull request merged**
